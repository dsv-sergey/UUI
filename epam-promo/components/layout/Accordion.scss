@use '../../assets/styles/fonts' as *;

$shadow: 0 3px 6px 0 rgba(29, 30, 38, 0.10), 0 1px 3px 0 rgba(29, 30, 38, 0.10);
$shadow-hovered: 0 6px 12px 0 rgba(29,30,38,0.10), 0 3px 6px 0 rgba(29,30,38,0.10);

.root {
    :global(.uui-accordion-toggler) {
        font-family: $font-sans-semibold;
        cursor: pointer;
    }

    :global(.uui-accordion-body) {
        font-family: $font-sans;
    }

    &.mode-block {
        :global(.uui-accordion-toggler) {
            background: var(--background-color);
            box-shadow: $shadow;
            fill: var(--icon-color);
            line-height: 24px;
            font-size: 18px;
            padding: 0 18px;

            :global(.uui-accordion-toggle-container) {
                min-height: 48px;
                padding-top: 5px;
                padding-bottom: 5px;
            }

            :global(.uui-accordion-title) {
                color: var(--text-color);
            }

            &:hover {
                box-shadow: $shadow-hovered;
            }
        }

        :global(.uui-accordion-body) {
            padding: 18px;
            line-height: 24px;
            letter-spacing: 0;
            background: var(--background-color);
            box-shadow: $shadow;
            border-top: 1px solid var(--border-color);
        }

        &:global(.uui-opened) {
            :global(.uui-accordion-toggler) {
                &:hover {
                    border-color: var(--border-color);
                }
            }
        }

        &:global(.uui-disabled) {
            :global(.uui-accordion-toggler) {
                cursor: default;
                border-color: var(--border-color-disabled);
                fill: var(--icon-color-disabled);

                :global(.uui-accordion-title) {
                    color: var(--text-color-disabled);
                }

                &:hover {
                    box-shadow: $shadow;
                }
            }
        }
    }

    &.mode-inline {
        :global(.uui-accordion-toggler) {
            border-bottom: 1px solid var(--border-color);
            fill: var(--icon-color);
            line-height: 18px;
            font-size: 14px;

            :global(.uui-accordion-toggle-container) {
                min-height: 24px;
                padding: 3px 12px 8px;
            }

            :global(.uui-accordion-title) {
                color: var(--text-color);
            }
        }

        :global(.uui-accordion-body) {
            padding: 1px 12px 12px;
            border-bottom: 1px solid var(--border-color);
        }

        &:global(.uui-opened) {
            :global(.uui-accordion-toggler) {
                border-bottom: unset;
            }
        }

        &:global(.uui-disabled) {
            :global(.uui-accordion-toggler) {
                cursor: default;
                border-color: var(--border-color-disabled);
                fill: var(--icon-color-disabled);
                border-bottom: unset;

                :global(.uui-accordion-title) {
                    color: var(--text-color-disabled);
                }
            }
        }
    }
}
