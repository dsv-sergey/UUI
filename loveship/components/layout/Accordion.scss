@use '../../assets/styles/scss/index'  as *;

.root {
    :global(.uui-accordion-toggler) {
        font-family: $font-sans-semibold;
        cursor: pointer;
    }

    :global(.uui-accordion-body) {
        font-family: $font-sans;
    }

    &.mode-block {
        border-radius: 6px;

        :global(.uui-accordion-toggler) {
            background: $white;
            box-shadow: 0 1px 2px 0 rgba(29, 30, 38, 0.1), 0 2px 10px 0 rgba(29, 30, 38, 0.15);
            border-radius: 6px;
            fill: $night600;
            line-height: 24px;
            font-size: 18px;
            padding: 0 24px;

            :global(.uui-accordion-toggle-container) {
                min-height: 48px;
                padding-top: 5px;
                padding-bottom: 5px;

                :global(.uui-accordion-title) {
                    color: $night700;
                }
            }

            &:hover {
                box-shadow: 0 4px 22px 0 rgba(29, 30, 38, 0.15);
            }
        }

        :global(.uui-accordion-body) {
            background: $white;
            padding: 18px 24px;
            line-height: 24px;
            letter-spacing: 0;
            border-top: unset;
            border-radius: 0 0 6px 6px;
        }

        &:global(.uui-opened) {
            box-shadow: 0 1px 2px 0 rgba(29, 30, 38, 0.1), 0 2px 10px 0 rgba(29, 30, 38, 0.15);

            :global(.uui-accordion-toggler) {
                border-bottom: 1px solid $night400;
                border-radius: 6px 6px 0 0;
                box-shadow: unset;

                &:hover {
                    box-shadow: unset;
                }
            }
        }

        &:global(.uui-disabled) {
            box-shadow: 0 1px 2px 0 rgba(29, 30, 38, 0.1), 0 2px 10px 0 rgba(29, 30, 38, 0.15);

            :global(.uui-accordion-toggler) {
                cursor: default;
                border-color: $night400;
                fill: $night500;
                border-bottom: 1px solid $night400;
                box-shadow: unset;

                :global(.uui-accordion-title) {
                    color: #9FA1AE;
                }
            }

            :global(.uui-opened) {
                border-radius: 6px 6px 0 0;
            }
        }
    }

    &.mode-inline {
        :global(.uui-accordion-toggler) {
            border-bottom: 1px solid $night400;
            fill: $night600;
            line-height: 18px;
            font-size: 14px;

            :global(.uui-accordion-toggle-container) {
                min-height: 24px;
                padding: 3px 12px 8px;

                :global(.uui-accordion-title) {
                    color: $night700;
                }
            }
        }

        :global(.uui-accordion-body) {
            padding: 1px 12px 12px;
            border-bottom: 1px solid $night400;
        }

        &:global(.uui-opened) {
            :global(.uui-accordion-toggler) {
                border-bottom: unset;
            }
        }

        &:global(.uui-disabled) {
            :global(.uui-accordion-toggler) {
                cursor: default;
                fill: $night500;

                :global(.uui-accordion-title) {
                    color: #9FA1AE;
                }
            }
        }
    }
}
