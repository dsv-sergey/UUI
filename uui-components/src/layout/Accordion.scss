.container {
    :global(.uui-accordion-toggle-container) {
        display: flex;
        align-items: center;
    }

    :global(.uui-accordion-title) {
        overflow: hidden;
        text-overflow: ellipsis;
    }

    :global(.uui-accordion-body) {
        overflow: hidden;
        text-overflow: ellipsis;
    }

    .spacer {
        flex: 1 1 auto;
    }
}
