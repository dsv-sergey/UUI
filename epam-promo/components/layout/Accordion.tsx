import { withMods } from '@epam/uui';
import { Accordion as uuiAccordion, AccordionProps } from '@epam/uui-components';
import { systemIcons } from '../../icons/icons';
import * as styles from '../../assets/styles/colorvars/layout/accordion-colorvars.scss';
import * as css from './Accordion.scss';

export interface AccordionMods {
    mode?: 'block' | 'inline';
}

function applyAccordionMods(mods: AccordionMods & AccordionProps) {
    return [
        css.root,
        styles.accordionColors,
        css['mode-' + (mods.mode || 'block')],
    ];
}

export const Accordion = withMods<AccordionProps, AccordionMods>(uuiAccordion, applyAccordionMods, (mods: AccordionMods & AccordionProps) => ({
    dropdownIcon: systemIcons[mods.mode === 'block' ? '60' : '30'].foldingArrow,
}));
