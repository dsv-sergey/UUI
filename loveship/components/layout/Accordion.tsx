import { withMods } from '@epam/uui';
import { Accordion as uuiAccordion, AccordionProps } from '@epam/uui-components';
import { systemIcons } from '../icons/icons';
import * as css from './Accordion.scss';

export interface AccordionMods {
    mode?: 'block' | 'inline';
}

function applyAccordionMods(mods: AccordionMods & AccordionProps) {
    return [
        css.root,
        css['mode-' + (mods.mode || 'block')]
    ];
}

export const Accordion = withMods<AccordionProps, AccordionMods>(uuiAccordion, applyAccordionMods, (mods: AccordionMods & AccordionProps) => ({
    dropdownIcon: systemIcons[mods.mode === 'block' ? '60' : '36'].foldingArrow,
}));
