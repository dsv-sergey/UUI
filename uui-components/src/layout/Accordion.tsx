import React from 'react';
import { IHasCX, IDisableable, uuiMod, IHasChildren, Icon, IEditable } from '@epam/uui';
import { IconContainer } from '../layout';
import * as css from './Accordion.scss';
import cx from 'classnames';

export interface AccordionProps extends Partial<IEditable<boolean>>, IHasCX, IDisableable, IHasChildren {
    title: string;
    dropdownIcon?: Icon;
}

export interface AccordionState {
    opened: boolean;
}

const uuiAccordion: Record<string, string> = {
    title: 'uui-accordion-title',
    toggler: 'uui-accordion-toggler',
    toggleContainer: 'uui-accordion-toggle-container',
    body: 'uui-accordion-body',
};

export class Accordion extends React.Component<AccordionProps, AccordionState> {
    state = {
        opened: this.props.value || false,
    };

    private toggleAccordion = () => {
        const opened = this.isOpened();
        this.props.onValueChange ? this.props.onValueChange(!opened) : this.setState({ opened: !opened });
    }

    isOpened = () => {
        return this.props.value !== undefined ? this.props.value : this.state.opened;
    }

    renderHeader = () => {
        const isAccordionOpened = this.isOpened();

        return (
            <div className={ cx(
                uuiAccordion.toggler,
                isAccordionOpened && uuiMod.opened,
                this.props.isDisabled && uuiMod.disabled,
            ) }
                 onClick={ !this.props.isDisabled ? this.toggleAccordion : undefined }>
                <div className={ cx(uuiAccordion.toggleContainer) }>
                    <div className={ cx(uuiAccordion.title) }>
                        { this.props.title }
                    </div>
                    <div className={ css.spacer } />
                    <IconContainer
                        icon={ this.props.dropdownIcon }
                        flipY={ isAccordionOpened }
                        cx={ this.props.isDisabled && uuiMod.disabled }
                    />
                </div>
            </div>
        );
    }

    renderBody = () => (
        <div className={ uuiAccordion.body }>
            { this.props.children }
        </div>
    )

    render() {
        const isAccordionOpened = this.isOpened();

        return (
            <div className={ cx(
                css.container,
                isAccordionOpened && !this.props.isDisabled && uuiMod.opened,
                this.props.isDisabled && uuiMod.disabled,
                this.props.cx,
            ) }>
                { this.renderHeader() }
                { this.props.children && isAccordionOpened ? this.renderBody() : null }
            </div>
        );
    }
}
