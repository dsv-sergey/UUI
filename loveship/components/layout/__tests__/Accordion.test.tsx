import React from "react";
import renderer from "react-test-renderer";
import {Accordion} from "../Accordion";
import * as acceptIcon from "../../icons/accept-12.svg";

describe("Accordion", () => {
    it("should be rendered correctly", () => {
        const tree = renderer
            .create(<Accordion title="test" />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("should be rendered correctly with extra props", () => {
        const tree = renderer
            .create(<Accordion
                title="test"
                mode="inline"
                dropdownIcon={ acceptIcon }
            />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});