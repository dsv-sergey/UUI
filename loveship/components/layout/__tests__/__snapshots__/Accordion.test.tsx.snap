// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion should be rendered correctly 1`] = `
<div
  className="container root mode-block"
>
  <div
    className="uui-accordion-toggler"
    onClick={[Function]}
  >
    <div
      className="uui-accordion-toggle-container"
    >
      <div
        className="uui-accordion-title"
      >
        test
      </div>
      <div
        className="spacer"
      />
      <div
        className="container uui-icon uui-enabled"
      >
        <svg
          className=""
          height={24}
          viewBox="0 0 12 24"
          width={12}
        >
          <use
            xlinkHref="#folding-arrow-18.svg"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`Accordion should be rendered correctly with extra props 1`] = `
<div
  className="container root mode-inline"
>
  <div
    className="uui-accordion-toggler"
    onClick={[Function]}
  >
    <div
      className="uui-accordion-toggle-container"
    >
      <div
        className="uui-accordion-title"
      >
        test
      </div>
      <div
        className="spacer"
      />
      <div
        className="container uui-icon uui-enabled"
      >
        <svg
          className=""
          height={24}
          viewBox="0 0 12 24"
          width={12}
        >
          <use
            xlinkHref="#folding-arrow-18.svg"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
